docs/getting-started/0-agent.md

```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
+You will need the genesis file of the Indy ledger you are connecting to, to accept/verify credentials.
+For example, lets say your agent will need to accept a verifiable credential from trinsic.id, you will probably need to download the genesis file for the Sovrin network.
+
+
+- [Sovrin Mainnet](https://github.com/sovrin-foundation/sovrin/blob/master/sovrin/pool_transactions_live_genesis)
+- [Sovrin Stagingnet](https://github.com/sovrin-foundation/sovrin/blob/master/sovrin/pool_transactions_sandbox_genesis)
+- [Sovrin Buildernet](https://github.com/sovrin-foundation/sovrin/blob/master/sovrin/pool_transactions_builder_genesis)
+
+More to find [here](https://github.com/sovrin-foundation/sovrin/tree/stable/sovrin)
+
+
+Other 
+- [Indicio TestNet](https://raw.githubusercontent.com/Indicio-tech/indicio-network/main/genesis_files/pool_transactions_testnet_genesis)
+
+#### Local network: 
+
+If you are going to test with a local network you have to download the genesis file corresponding to the networks local ledger.
+Example: [DTS Verifiable Credential Issuer Service](https://github.com/bcgov/dts-vc-issuer-service)
+Corresponding genesis file: http://test.bcovrin.vonx.io/genesis
+
+Code for downloading genesis file
+
+```ts
+//Function for downloading genesis file and return string value
+async function downloadGenesis(genesisUrl: string) {
+  try {
+    const response = await axios.get(genesisUrl); //Axios is a lightweight HTTP client, you can use any other library
+    return response.data;
+  } catch (e) {
+    console.log('EXCEPTION> downloadGenesis:', e);
+    return null;
+  }
+}
+
+
+//Function for storing genesis string value to a local file on the device 
+async function storeGenesis(
+  genesis: string,
+  fileName: string,
+): Promise<string> {
+  try {
+    const genesisPath = `${RNFS.DocumentDirectoryPath}/${fileName}`;
+    await RNFS.writeFile(genesisPath, genesis, 'utf8');
+    return genesisPath;
+  } catch (e) {
+    console.log('EXCEPTION> storeGenesis:', e);
+    return '';
+  }
+}
+
+//Somewhere in the code (before init Agent)
+const genesis = await downloadGenesis(GENESIS_URL);
+const genesisPath = await storeGenesis(genesisString, 'genesis.txn');
+
+```
+
+Note: You do not need the genesis file if you are creating a connection between your Agent and another Agent for exchanging simple messages.
+
+### 3- Get Mediator Connection URL (Optional)
+
+Mediators (Routing Agents) are Agents that serve as intermediaries to facilitate the flow of messages between other types of agents.
+You will need a mediator Agent if you are going to deal with VC (Verifiable Credentials), however, you can ignore the mediator step if you are creating an Agent for the sole purpose of exchanging messages with another Agent. 
```

Comment:
This assumes mobile environments. Mediators are only needed when you can't host an endpoint to receive messages yourself (this is the case on mobile environments). It is required for all exchanges on mobile, not only when dealing with VCs